### PR TITLE
fix linux compile

### PR DESCRIPTION
**Pull request by psychocrypt**
The current code is not compiling under linux with gcc 5.4.0, this PR fixes the compile issues.

- fix wrong type for function call
- remove unknown function

### CODE REVIEW

File: crypto/cryptonight_aesni.h [hunk 1]
```diff
@@ -327,7 +327,6 @@ void cryptonight_hash(const void* input, size_t len, void* output, cryptonight_c
 	__m128i division_result_xmm = _mm_cvtsi64_si128(h0[12]);
 	__m128i sqrt_result_xmm = _mm_cvtsi64_si128(h0[13]);
 
-	_control87(RC_DOWN, MCW_RC);
```
Comment (SChernykh) [inline crypto/cryptonight_aesni.h:330]: You can't just remove _control87, it sets FPU rounding mode to "round down" aka "round to negative infinity". It must be replaced with `std::fesetround(FE_DOWNWARD);`

File: crypto/cryptonight_aesni.h [hunk 2]
```diff
@@ -300,7 +300,7 @@ static __forceinline __m128i int_sqrt33_1_double_precision(const uint64_t n0)
 
 	uint64_t x2 = (s - (1022ULL << 32)) * (r - s - (1022ULL << 32) + 1);
 	//if (x2 < n0) ++r;
-	_addcarry_u64(_subborrow_u64(0, x2, n0, &x2), r, 0, &r);
+	_addcarry_u64(_subborrow_u64(0, x2, n0, (unsigned long long int*)&x2), r, 0, (unsigned long long int*)&r);
```
Comment (SChernykh) [inline crypto/cryptonight_aesni.h:303]: It's better to write uint64_t here just like in the rest of the function.

**Review by SChernykh — CHANGES_REQUESTED** [review]
Please set correct FPU rounding mode before the main loop

File: crypto/cryptonight_aesni.h (hunk 2 above)
Comment (SChernykh) [reply crypto/cryptonight_aesni.h:303]: Or not... I guess it requires unsigned long long int here to compile successfully on gcc.

Comment (psychocrypt) [reply crypto/cryptonight_aesni.h:303]: Yes `uint64_t*` is not on all system `unsigned long long int*`

**Comment by SChernykh** [comment]
Not needed anymore, it's already fixed.

**Pull request closed without merging**